### PR TITLE
WP-4877 Update Disposable usage recommendation

### PR DESCRIPTION
**Pull request by georgelesica-wf**
> Please apply the appropriate label to your PR:
> - bug / critical
> - improvement / optimization / tech-debt / UX
> - feature

### Description

The `Disposable` recommendation was outdated.

### Changes

Update it.

### Semantic Versioning

- [x] **Patch**
  - [x] This change does not affect the public API
  - [ ] This change fixes existing incorrect behavior without any additions
- [ ] **Minor**
  - [ ] This change adds something to the public API
  - [ ] This change deprecates a part of the public API
* [ ] Major
  - [ ] This change modifies part of the public API in a backwards-incompatible manner
  - [ ] This change removes part of the public API

### Testing/QA

- [ ] CI passes

### Code Review

@dustinlessard-wf
@evanweible-wf
@jayudey-wf
@maxwellpeterson-wf
@sebastianmalysa-wf
@trentgrover-wf

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Raven
<details>
<summary>Number of Findings: 0</summary>

</details>

File: lib/src/common/disposable.dart [hunk 1]
```diff
@@ -126,10 +126,14 @@ typedef Future<dynamic> Disposer();
...
-/// fine but the first is the simplest and most powerful. Using the class
-/// as an interface will require significant effort.
+/// We recommend consuming this class in one of two ways.
+///
+///   1. As a base class
+///   2. As a concrete proxy
+///
+/// We do not recommend using this class as a mixin or as an interface.
+/// Use as a mixin can cause the `onDispose` method to be shadowed. Use
+/// as an interface is just cumbersome.
```
Comment (aaronstgeorge-wf) [inline lib/src/common/disposable.dart]: It might be valuable to say something about, `if an interface is required use DispoablabeManagerV777` here.

Comment (georgelesica-wf) [inline lib/src/common/disposable.dart]: Dun

**Comment by todbachman-wf** [comment]
+10. Confirmed CI passed.

**Comment by todbachman-wf** [comment]
QA +1

- [x] Testing instruction
- [x] Dev +1's
- [x] Dev/QA +10 with detail of what was tested
- [ ] Security review (if needed)
  - N/A
- [ ] Unit tests created/updated 
  - N/A
- [x] All unit tests pass
- [x] Confirm there is a ticket in the PR title
- [x] Rosie has run and reports properly the release the ticket will be included in
- [x] Rosie reports a clean dependency scan

Merging into master

**Pull request merged**